conversion.go

```diff
@@ -46,21 +46,62 @@ func (z *Int) ToBig() *big.Int {
 	if z == nil {
 		return nil
 	}
-	b := new(big.Int)
+	var b *big.Int
+	z.IntoBig(&b)
+	return b
+}
+
+// IntoBig sets a provided big.Int to the value of z.
+// Sets `nil` if z is nil (thus the double pointer).
+func (z *Int) IntoBig(b **big.Int) {
```

Comment:
I don't like to have a double-pointer in the API. It's not nice to work with, because it's so unusual. 
I can be persuaded otherwise, but, why not just pick one of these options:
1. panic on nil deref if `z` is nil
2. Set the `b` to zero if `z` is nil

And then get rid of the `**`?

Reply:
Well, the purpose is to convert between uint256 and big.Int. If nil is a valid uint256 value, it we should be able to convert it to big.Int. IMO we''ll shoot ourselves in the foot with all the optional uint256 fields if we panic all of a sudden. Also converting an unset-optional into a set-0 seems bad.

Reply:
**big.Int is similar to *[] FWIW. Maybe unusual to see a double pointer like that, but it's not a unique construct. You can definitely work around it by delegating nil checks to outer code, but then half the conversion happens inside the lib, but the other half (allocating/nilling the big.Int happens outside).

IMO the API "strageness" isn't that bad compared to having everyone become responsible for handling nil/non-nil in their code explicitly.

Reply:
Point in case, straight from the Go standard library: https://github.com/golang/go/blob/b8f83e22703ee23d49d95154449ce7066402d5c9/src/crypto/internal/boring/boring.go#L83

Converting big ints by placing them into an existing variable :)

Reply:
Ok!

Reply:
But if `b` is nil, this _will_ panic. Is that intentional? I guess it is, just want to double-check that's what we want

Reply:
Apart from sending in nil explicitly, there's no meaningful way you can end up with nil.

i.e. if you have a nil big.int pointer, you can still take it's address and it will be a non-nil pointer you can use to init the big.int. Since nobody's passing double pointers around in general for big.ints, you can't realistically end up with an accidental nil double pointer.